jgrapht-core/src/main/java/org/jgrapht/alg/util/DirectedNeighborIndex.java

```diff
@@ -40,12 +39,15 @@
  *
  * @author Charles Fry
  * @since Dec 13, 2005
+ * 
+ * @deprecated Use {@link org.jgrapht.alg.util.NeighborCache} instead.
```

Comment:
Don't move this deprecated class; it should stay in the original location since that's where existing apps are referencing it.

Reply:
I moved back the deprecated classes to `org.jgrapht.alg` package. Since the new implementation is in another package, how should I go about making the nested static class *NeighborIndex.Neighbors* accessible to it?  
I only see two scenarios:
1. keep the class nested but make it public  
1. make it a top-level class

Reply:
In general we use the following strategy when moving existing classes to a new package:
1. Move the original classes to the new package, but leave a copy of the original classes behind
2. Make the original classes deprecated.
Does this resolve your issue?

Reply:
I believe it does @jkinable , but since the new implementation doesn't rely on the original classes - only on one of its nested class - I thought we are going to get rid of the original classes somehow in one go.